script/gn-check.js

```diff
@@ -0,0 +1,23 @@
+const sh = require('shelljs')
+const path = require('path')
+const fs = require('fs')
+
+const ELECTRON_DIR = path.resolve(__dirname, '..', '..')
+let GN_CHECK_DIR
+
+if (process.env.ELECTRON_OUT_DIR) {
+  GN_CHECK_DIR = `../out/${process.env.ELECTRON_OUT_DIR}`
+} else {
+  for (const buildType of ['Debug', 'Testing', 'Release']) {
+    const outPath = path.resolve(ELECTRON_DIR, 'out', buildType)
+    if (fs.existsSync(outPath)) {
+      GN_CHECK_DIR = `../out/${buildType}`
+      break
+    }
+  }
+}
+
+const gnCheckString = `gn check ${GN_CHECK_DIR} //electron:electron_lib`
+if (sh.exec(gnCheckString).code !== 0) {
+  sh.exit(1)
+}
```

Comment:
Not a blocker for this use case, but as a rule-of-thumb it's better to pass the exit code along -- it's not uncommon for apps to return various codes to signal different errors.

Consider a hypothetical case where this script is called by another script that wants to do different things depending on the error code. We wouldn't want all nonzero codes to be mapped to 1.

```js
sh.exit(sh.exec(gnCheckString).code)
```